### PR TITLE
samples: drivers: Enable SPI MEMORY driver samples for nrfl54l15

### PR DESCRIPTION
**Pull request by nordic-bami**
Cover more SPI flash memeory driver api methods
Extend testing for nrf54l15

### CODE REVIEW

**Comment by nordic-bami** [comment]
Please review

**Review by kartben — CHANGES_REQUESTED** [review]
Coverage should be a goal for a test, not a user-facing code sample
https://docs.zephyrproject.org/latest/samples/sample_definition_and_criteria.html

Thanks

**Comment by nordic-bami** [comment]
> Coverage should be a goal for a test, not a user-facing code sample https://docs.zephyrproject.org/latest/samples/sample_definition_and_criteria.html
> 
> Thanks

Agree.
I need to wait until upmerge finishes on our side before modifying the test case.

**Review by nordic-segl — APPROVED** [review]

**Review by nordic-babu — APPROVED** [review]

**Review by nordic-piks — APPROVED** [review]

File: samples/drivers/jesd216/boards/nrf54l15pdk_nrf54l15_cpuapp.overlay [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&mx25r64 {
+	status = "okay";
+};
+
+/ {
+	aliases {
+		jedec-spi-nor = &mx25r64;
```
Comment (kartben) [inline samples/drivers/jesd216/boards/nrf54l15pdk_nrf54l15_cpuapp.overlay:13]: what are you trying to accomplish with this alias?

Comment (kartben) [reply samples/drivers/jesd216/boards/nrf54l15pdk_nrf54l15_cpuapp.overlay:13]: @nordic-bami ping

Comment (nordic-bami) [reply samples/drivers/jesd216/boards/nrf54l15pdk_nrf54l15_cpuapp.overlay:13]: Hmm i have replied but it seems to not be visible:

DT_ macro in the test code uses such name

#if DT_HAS_COMPAT_STATUS_OKAY(jedec_spi_nor)

Comment (kartben) [reply samples/drivers/jesd216/boards/nrf54l15pdk_nrf54l15_cpuapp.overlay:13]: You are mixing up concepts. The `mx25r64` node already has the compatible so you really only need to enable it. https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/nordic/nrf54l15pdk/nrf54l15_cpuapp_common.dtsi#L137-L138

Please just drop these aliases, they are not correct/useless.

Comment (nordic-bami) [reply samples/drivers/jesd216/boards/nrf54l15pdk_nrf54l15_cpuapp.overlay]: Updated

**Review by kartben — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**